### ISSUE TITLE
Bad validation message

### ISSUE THREAD

**Issue by jma** [ngx-formly/ngx-formly#3914]
**Description**
A wrong validation message is displayed with arrays.

**Minimal Reproduction**

Here you can find a stackblitz reproduction example. If you click on submit we got a required error message instead of a minLength. Let us know if we need to open an issue. 

https://stackblitz.com/edit/angular-kwmyzs-pmxgrs?file=src%2Fassets%2Fjson-schema%2Farrays.json

Thanks in advance.

**Comment by aitboudad** [comment]
it is required to pass an empty string for that case, if I'm not wrong. Please try the following solution instead:
```patch
"items": {
  "type": "string",
  "minLength": 0,
+ "default": ""
}
```

**Comment by jma** [comment]
Unfortunately, this does not work. Here is a modified example:

https://stackblitz.com/edit/angular-kwmyzs-pmxgrs?file=src%2Fassets%2Fjson-schema%2Farrays.json

**Comment by jma** [comment]
In fact a field with an parent field of type array should never be required at least for a JSONScheam. The required property is only available on the object type thus only for the children of an object field. I tried to add a patch in the prePopulate of my extension, but changing the `required` property of a field change also the `required` property of the children...

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.3.4](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.3.4) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by jma** [comment]
> This issue has been fixed and released as part of [v6.3.4](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.3.4) release.
> 
> Please let us know, in case you are still encountering a similar issue/problem. Thank you!

Unfortunately this is the case. You have introduce a new bug in the version 6.3.4. Using the following stackblitz example the field "new" is not marked as required anymore:
https://stackblitz.com/edit/angular-kwmyzs-1aw4z2?file=src%2Fassets%2Fjson-schema%2Farrays.json

Thanks in advance.

**Comment by aitboudad** [comment]
> https://stackblitz.com/edit/angular-kwmyzs-1aw4z2?file=src%2Fassets%2Fjson-schema%2Farrays.json

The issue here is when/in which case an array item is considered required. here is how it is handled:
- Object is underlined, name field is not required
- Object set, name field is required

**Comment by jma** [comment]
> The issue here is when/in which case an array item is considered required. here is how it is handled:
> 
> * Object is underlined, name field is not required
> * Object set, name field is required

If I understood well, you want to make a input field required only when a field is touch, but we have several problems here. Running the following modified example:
https://stackblitz.com/edit/angular-kwmyzs-1aw4z2?file=src%2Fassets%2Fjson-schema%2Farrays.json

1. after loading the editor is ok
2. after clicking on "submit" the validation message is wrong: the array is not required but need at least one item, thus the error message "This field is required" is wrong.
3. now if a new item is added, we can see that the form is valid which is not true, if we try to validate the model against the JOSONSchema, the "new" property is missing as it is required.
4. If I enter a "space" in the "new" field and suppress it (the field is touched), the message "This field is required" appears but I cannot go back to the situation before I touched the field.

**Comment by aitboudad** [comment]
The proper fix still needs more time, in the meantime please try the following workaround:

> 2.after clicking on "submit" the validation message is wrong: the array is not required but need at least one item, thus the error message "This field is required" is wrong.

Remove "listOfStrings" from required properties since it'll be validated by `minItems`:

```patch
- "required": ["listOfStrings", "requiredField"],
+ "required": ["requiredField"],
```

> 3. now if a new item is added, we can see that the form is valid which is not true, if we try to validate the model against the JOSONSchema, the "new" property is missing as it is required.

set default as object in order enforce the validation

```patch
{
  "title": "value",
  "type": "object",
  "required": ["new"],
+ "default": {},
  "properties": {
    "new": {
      "title": "New",
      "type": "string"
    }
  }
}
```

**Comment by jma** [comment]
Thank you, we will wait until the issue will be fixed. We have huge JSONSchemas and we do not want to change it.

Thanks again.

**Comment by aitboudad** [comment]
I've pushed a fix in [v6.3.5](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.3.5) release.
For the validation message, I think it should be handled from your side instead.
